cuDNN doesn't seem to work on Windows 8.1

I am able to train on GPU. But I don't seem to be able to run cuDNN on Windows 8.1.

I downloaded 0.4 examples, imported into Eclipse (with java 1.7 x64), and made only those 2 changes:

In the parent pom.xml:
`<nd4j.backend>nd4j-cuda-7.5</nd4j.backend>`

in the dl4j-examples pom.xml added this in the dependencies section:
`		<dependency>`
`		    <groupId>org.deeplearning4j</groupId>`
`		    <artifactId>deeplearning4j-cuda-7.5</artifactId>`
`		    <version>${dl4j.version}</version>`
`		</dependency>                 `

This is what I see when running Lenet example:

o.d.e.c.LenetMnistExample - Load data....


o.d.e.c.LenetMnistExample - Build model....
o.d.n.l.c.ConvolutionLayer - Could not load CudnnConvolutionHelper
java.lang.UnsatisfiedLinkError: no jnicuda in java.library.path
	at java.lang.ClassLoader.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at java.lang.Runtime.loadLibrary0(Unknown Source) ~[na:1.7.0_79]
	at java.lang.System.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:726) ~[javacpp-1.2.2.jar:1.2.2]
	at org.bytedeco.javacpp.Loader.load(Loader.java:501) ~[javacpp-1.2.2.jar:1.2.2]
Caused by: java.lang.UnsatisfiedLinkError: no cuda in java.library.path
	at java.lang.ClassLoader.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at java.lang.Runtime.loadLibrary0(Unknown Source) ~[na:1.7.0_79]
	at java.lang.System.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:726) ~[javacpp-1.2.2.jar:1.2.2]
	at org.bytedeco.javacpp.Loader.load(Loader.java:492) ~[javacpp-1.2.2.jar:1.2.2]
o.d.n.l.c.s.SubsamplingLayer - Could not load CudnnSubsamplingHelper
java.lang.NoClassDefFoundError: Could not initialize class org.bytedeco.javacpp.cudnn
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.createHandles(CudnnSubsamplingHelper.java:89) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.<init>(CudnnSubsamplingHelper.java:76) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper.<init>(CudnnSubsamplingHelper.java:105) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:1.7.0_79]
o.d.n.l.c.ConvolutionLayer - Could not load CudnnConvolutionHelper
java.lang.NoClassDefFoundError: Could not initialize class org.bytedeco.javacpp.cudnn
	at org.deeplearning4j.nn.layers.convolution.CudnnConvolutionHelper$CudnnContext.createHandles(CudnnConvolutionHelper.java:94) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.CudnnConvolutionHelper$CudnnContext.<init>(CudnnConvolutionHelper.java:78) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.CudnnConvolutionHelper.<init>(CudnnConvolutionHelper.java:136) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:1.7.0_79]
o.d.n.l.c.s.SubsamplingLayer - Could not load CudnnSubsamplingHelper
java.lang.NoClassDefFoundError: Could not initialize class org.bytedeco.javacpp.cudnn
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.createHandles(CudnnSubsamplingHelper.java:89) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.<init>(CudnnSubsamplingHelper.java:76) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper.<init>(CudnnSubsamplingHelper.java:105) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:1.7.0_79]
o.d.e.c.LenetMnistExample - Train model....
o.d.o.l.ScoreIterationListener - Score at iteration 0 is 2.5576238169173227
o.d.o.l.ScoreIterationListener - Score at iteration 1 is 2.414856652833581

I upgraded cudnn from v4 to v5. Installed v5 for Cuda 7.5 (not RC), but still see the same errors.

Note: cudnn (v4) did work for me with Theano.

@AlexDBlack, 
c:\>where cudnn64_5.dll
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.5\bin\cudnn64_5.dll

I am attaching the stack trace anyway (just in case you find any difference):

o.d.e.c.LenetMnistExample - Load data....


o.d.e.c.LenetMnistExample - Build model....
o.d.n.l.c.ConvolutionLayer - Could not load CudnnConvolutionHelper
java.lang.UnsatisfiedLinkError: no jnicuda in java.library.path
	at java.lang.ClassLoader.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at java.lang.Runtime.loadLibrary0(Unknown Source) ~[na:1.7.0_79]
	at java.lang.System.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:726) ~[javacpp-1.2.2.jar:1.2.2]
	at org.bytedeco.javacpp.Loader.load(Loader.java:501) ~[javacpp-1.2.2.jar:1.2.2]
Caused by: java.lang.UnsatisfiedLinkError: no cuda in java.library.path
	at java.lang.ClassLoader.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at java.lang.Runtime.loadLibrary0(Unknown Source) ~[na:1.7.0_79]
	at java.lang.System.loadLibrary(Unknown Source) ~[na:1.7.0_79]
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:726) ~[javacpp-1.2.2.jar:1.2.2]
	at org.bytedeco.javacpp.Loader.load(Loader.java:492) ~[javacpp-1.2.2.jar:1.2.2]
o.d.n.l.c.s.SubsamplingLayer - Could not load CudnnSubsamplingHelper
java.lang.NoClassDefFoundError: Could not initialize class org.bytedeco.javacpp.cudnn
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.createHandles(CudnnSubsamplingHelper.java:89) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.<init>(CudnnSubsamplingHelper.java:76) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper.<init>(CudnnSubsamplingHelper.java:105) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:1.7.0_79]
o.d.n.l.c.ConvolutionLayer - Could not load CudnnConvolutionHelper
java.lang.NoClassDefFoundError: Could not initialize class org.bytedeco.javacpp.cudnn
	at org.deeplearning4j.nn.layers.convolution.CudnnConvolutionHelper$CudnnContext.createHandles(CudnnConvolutionHelper.java:94) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.CudnnConvolutionHelper$CudnnContext.<init>(CudnnConvolutionHelper.java:78) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.CudnnConvolutionHelper.<init>(CudnnConvolutionHelper.java:136) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:1.7.0_79]
o.d.n.l.c.s.SubsamplingLayer - Could not load CudnnSubsamplingHelper
java.lang.NoClassDefFoundError: Could not initialize class org.bytedeco.javacpp.cudnn
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.createHandles(CudnnSubsamplingHelper.java:89) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper$CudnnContext.<init>(CudnnSubsamplingHelper.java:76) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at org.deeplearning4j.nn.layers.convolution.subsampling.CudnnSubsamplingHelper.<init>(CudnnSubsamplingHelper.java:105) ~[deeplearning4j-cuda-7.5-0.4.0.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:1.7.0_79]
o.d.e.c.LenetMnistExample - Train model....
o.d.o.l.ScoreIterationListener - Score at iteration 0 is 2.5576238169173227
o.d.o.l.ScoreIterationListener - Score at iteration 1 is 2.414856652833581
o.d.o.l.ScoreIterationListener - Score at iteration 2 is 2.7043355067739876
o.d.o.l.ScoreIterationListener - Score at iteration 3 is 2.2946867901265167
o.d.o.l.ScoreIterationListener - Score at iteration 4 is 1.8114470449680762

@saudet - after I added this line, the example does not run at all, this is the stack trace:

Exception in thread "main" java.lang.UnsatisfiedLinkError: no jnicuda in java.library.path
	at java.lang.ClassLoader.loadLibrary(Unknown Source)
	at java.lang.Runtime.loadLibrary0(Unknown Source)
	at java.lang.System.loadLibrary(Unknown Source)
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:726)
	at org.bytedeco.javacpp.Loader.load(Loader.java:501)
	at org.bytedeco.javacpp.Loader.load(Loader.java:418)
	at org.bytedeco.javacpp.cuda.<clinit>(cuda.java:10)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Unknown Source)
	at org.bytedeco.javacpp.Loader.load(Loader.java:473)
	at org.bytedeco.javacpp.Loader.load(Loader.java:434)
	at org.deeplearning4j.examples.convolution.LenetMnistExample.main(LenetMnistExample.java:36)
Caused by: java.lang.UnsatisfiedLinkError: no cuda in java.library.path
	at java.lang.ClassLoader.loadLibrary(Unknown Source)
	at java.lang.Runtime.loadLibrary0(Unknown Source)
	at java.lang.System.loadLibrary(Unknown Source)
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:726)
	at org.bytedeco.javacpp.Loader.load(Loader.java:492)
	... 7 more

--- a/deeplearning4j-cuda-7.5/pom.xml
+++ b/deeplearning4j-cuda-7.5/pom.xml
@@ -25,6 +25,11 @@
         <version>0.4.1-SNAPSHOT</version>
     </parent>
 
+    <properties>
+        <cuda.version>7.5</cuda.version>
+        <javacpp.version>1.2</javacpp.version>
+    </properties>
+
     <dependencyManagement>
         <dependencies>
             <dependency>
@@ -34,7 +39,7 @@
             </dependency>
             <dependency>
                 <groupId>org.nd4j</groupId>
-                <artifactId>nd4j-cuda-7.5</artifactId>
+                <artifactId>nd4j-cuda-${cuda.version}</artifactId>
                 <version>${nd4j.version}</version>
             </dependency>
         </dependencies>
@@ -51,7 +56,7 @@
         </dependency>
         <dependency>
             <groupId>org.nd4j</groupId>
-            <artifactId>nd4j-cuda-7.5</artifactId>
+            <artifactId>nd4j-cuda-${cuda.version}</artifactId>
         </dependency>
         <dependency>
             <groupId>org.deeplearning4j</groupId>
@@ -61,7 +66,31 @@
         <dependency>
             <groupId>org.bytedeco.javacpp-presets</groupId>
             <artifactId>cuda</artifactId>
-            <version>7.5-1.2</version>
+            <version>${cuda.version}-${javacpp.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>cuda</artifactId>
+            <version>${cuda.version}-${javacpp.version}</version>
+            <classifier>linux-x86_64</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>cuda</artifactId>
+            <version>${cuda.version}-${javacpp.version}</version>
+            <classifier>macosx-x86_64</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>cuda</artifactId>
+            <version>${cuda.version}-${javacpp.version}</version>
+            <classifier>windows-x86_64</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.bytedeco.javacpp-presets</groupId>
+            <artifactId>cuda</artifactId>
+            <version>${cuda.version}-${javacpp.version}</version>
+            <classifier>linux-ppc64le</classifier>
         </dependency>
         <dependency>
             <groupId>junit</groupId>

